paddle/fluid/inference/api/demo_ci/CMakeLists.txt

```diff
@@ -104,7 +104,7 @@ if(WITH_MKL)
     if(WIN32)
       set(MKLDNN_LIB ${MKLDNN_PATH}/lib/mkldnn.lib)
     else(WIN32)
-      set(MKLDNN_LIB ${MKLDNN_PATH}/lib/libmkldnn.so.0)
+      set(MKLDNN_LIB ${MKLDNN_PATH}/lib/libmkldnn.so.1)
```

Comment:
> Some example CMakeLists.txt of typical customer app would also be useful.

Here is the inference demo CMakeLists.txt. We don't want users to change the name from `libmkldnn.so.0` to `libmkldnn.so.1`. Could you rename it when used in PaddlePaddle? @jczaja

Reply:
But what you mean by renaming? When you will build paddle after this PR is merged, this will produce libmkldnn.so.1 which is needed to run MKL-DNN 1.0 code. If you will build Paddle with MKL-DNN 1.0 and try to run it with old libmkldnn.so.0, name doesn't matters, it still won't work. These 2 libs are ABI incompatibile.

Reply:
> These 2 libs are ABI incompatibile.

For PaddlePaddle users, they don't care ABI incompatibile of mkldnn.

>  If you will build Paddle with MKL-DNN 1.0 and try to run it with old libmkldnn.so.0

For Paddle inference library, there should be only one mkldnn library named `libmkldnn.so.0`. Thus, in the packaged whl and inference library, we want the name of mkldnn library be the same as before. Otherwise, the uses should modify their CMakeLists.txt.

Reply:
So as far as I understand that, users are linking directly to libmkldnn.so.0 ? They should just link to libmkldnn.so which will symbolicly link to any needed Major name.

Reply:
> They should just link to libmkldnn.so which will symbolicly link to any needed Major name.

Maybe it's a history problem. Now the inference library package the `libmkldnn.so.0`. https://www.paddlepaddle.org.cn/documentation/docs/en/advanced_usage/deploy/inference/build_and_install_lib_en.html You can download the `fluid_inference.tgz`
![image](https://user-images.githubusercontent.com/6836917/69320084-7912b000-0c7b-11ea-9912-694b98557672.png)

Reply:
There is still `libmkldnn.so.1`?

Reply:
Sorry, I've messed something up during rebasing process, fixing it right now